src/OpenLoco/GameCommands/RenameCompanyOwner.cpp

```diff
@@ -0,0 +1,121 @@
+#include "../CompanyManager.h"
+#include "../Economy/Expenditures.h"
+#include "../Interop/Interop.hpp"
+#include "../Limits.h"
+#include "../Localisation/FormatArguments.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Localisation/StringManager.h"
+#include "../Types.hpp"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    void sub_434BA1();
+
+    /**
+     * 0x00434914
+     * Renames a particular company owner.
+     *
+     * This command is called 3 times before the buffer is applied. Each time, 12 chars of the 36 char buffer are provided.
+     * The resulting company owner name has a maximum length of 31 chars; the last bytes are not used.
+     *
+     * @param flags @<bl> - game command flags
+     * @param companyId @<cl> - company id
+     * @param index @<ax> - update index (in order of: 1, 2, 0)
+     * @param buffer0 @<edx> - First part (4 chars) of the 12 update buffer
+     * @param buffer1 @<dx> - Second part (4 chars) of the 12 update buffer
+     * @param buffer2 @<bp> - Third part (4 chars) of the 12 update buffer
+     * @return @<ebx> - returns 0 if rename is successful; otherwise GameCommands::FAILURE
+     */
+    static uint32_t changeCompanyOwnerName(const uint8_t flags, CompanyId companyId, int16_t index, uint32_t buffer0, uint32_t buffer1, uint32_t buffer2)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
```

Comment:
parts of this method are the same as the changeCompanyName function, any way to  combine them to reduce duplicate code?

Reply:
all the name changes are similar

Reply:
Yeah, they were all based on each other. Perhaps we could extract some common components, though ultimately we're likely to refactor all of these game commands into classes, anyway.